src/Compilers/CSharp/Test/Syntax/Parsing/WithElementParsingTests.cs

```diff
@@ -0,0 +1,2336 @@
+                    }
+                }
+                N(SyntaxKind.CloseBracketToken);
+            }
+            EOF();
+        }
+    }
+
+    [Theory, MemberData(nameof(CollectionArgumentsLanguageVersions))]
+    public void WithElement39(LanguageVersion languageVersion)
+    {
+        UsingTree("""
+            void M()
+            {
+                var v = [await with()];
+            }
+            """,
+            TestOptions.Regular.WithLanguageVersion(languageVersion),
+            // (3,20): error CS1003: Syntax error, ',' expected
+            //     var v = [await with()];
+            Diagnostic(ErrorCode.ERR_SyntaxError, "with").WithArguments(",").WithLocation(3, 20));
+
+        N(SyntaxKind.CompilationUnit);
+        {
+            N(SyntaxKind.GlobalStatement);
+            {
+                N(SyntaxKind.LocalFunctionStatement);
+                {
+                    N(SyntaxKind.PredefinedType);
+                    {
+                        N(SyntaxKind.VoidKeyword);
+                    }
+                    N(SyntaxKind.IdentifierToken, "M");
+                    N(SyntaxKind.ParameterList);
+                    {
+                        N(SyntaxKind.OpenParenToken);
+                        N(SyntaxKind.CloseParenToken);
+                    }
+                    N(SyntaxKind.Block);
+                    {
+                        N(SyntaxKind.OpenBraceToken);
+                        N(SyntaxKind.LocalDeclarationStatement);
+                        {
+                            N(SyntaxKind.VariableDeclaration);
+                            {
+                                N(SyntaxKind.IdentifierName);
+                                {
+                                    N(SyntaxKind.IdentifierToken, "var");
+                                }
+                                N(SyntaxKind.VariableDeclarator);
+                                {
+                                    N(SyntaxKind.IdentifierToken, "v");
+                                    N(SyntaxKind.EqualsValueClause);
+                                    {
+                                        N(SyntaxKind.EqualsToken);
+                                        N(SyntaxKind.CollectionExpression);
+                                        {
+                                            N(SyntaxKind.OpenBracketToken);
+                                            N(SyntaxKind.ExpressionElement);
+                                            {
+                                                N(SyntaxKind.IdentifierName);
+                                                {
+                                                    N(SyntaxKind.IdentifierToken, "await");
+                                                }
+                                            }
+                                            M(SyntaxKind.CommaToken);
+                                            CollectionArgumentsOrInvocation(languageVersion);
```

Comment:
Is this expected? I thought that with a prefix `await`, that this would parse as an invocation always.

Reply:
in this case, because we're not in an async block, the `await` is itself an expression (not the start of an async-expression).  so this is parsed as `[await , with()]` (just with a missing comma).  See the next test for an async context.